From 2b902fc095de90fcc68900bf80b236ac54908a9c Mon Sep 17 00:00:00 2001
From: Paul Seyfert <paul.seyfert@mib.infn.it>
Date: Sat, 29 Oct 2016 00:48:12 +0200
Subject: [PATCH] adapt to interface change in
 root-mirror/root@0ae4aaab01c42a5d5dfb80bdd0ab7f95012e99d6

use ROOT_VERSION_CODE and ROOT_VERSION macro to remain backward
compatible
---
 include/MethodFastBDT.h | 20 +++++++++++++++++---
 src/MethodFastBDT.cxx   | 14 ++++++++++++++
 2 files changed, 31 insertions(+), 3 deletions(-)

diff --git a/include/MethodFastBDT.h b/include/MethodFastBDT.h
index 06e4fed..eaebdad 100644
--- a/include/MethodFastBDT.h
+++ b/include/MethodFastBDT.h
@@ -7,7 +7,9 @@
 #ifndef ROOT_TMVA_MethodBase
 #include "TMVA/MethodBase.h"
 #endif
-
+#ifndef ROOT_RVersion
+#include "RVersion.h"
+#endif
 
 #include "FastBDT.h"
 
@@ -27,23 +29,35 @@ namespace TMVA {
        * @param methodTitle title of the method
        * @param theData data which was added to the TMVA Factory
        * @param theOption options string passed by the user to this method
-       * @param theTargetDir target directory in ROOT file which stores the information about the training 
+       * @param theTargetDir target directory in ROOT file which stores the information about the training (removed in ROOT 6.08.00)
        */
+#if ROOT_VERSION_CODE >= ROOT_VERSION(6,8,0)
+      MethodFastBDT( const TString& jobName,
+                 const TString& methodTitle,
+                 DataSetInfo& theData,
+                 const TString& theOption = "");
+#else
       MethodFastBDT( const TString& jobName,
                  const TString& methodTitle,
                  DataSetInfo& theData,
                  const TString& theOption = "",
                  TDirectory* theTargetDir = 0 );
+#endif
 
       /**
        * Constructor for calculating BDT-MVA using previously generated decision trees
        * @param theData data which should be classified
        * @param theWeightFile the xml file from which the method reads the weights
-       * @param theTargetDir target directory in ROOT file which stores information about the training
+       * @param theTargetDir target directory in ROOT file which stores information about the training (removed in ROOT 6.08.00)
        */
+#if ROOT_VERSION_CODE >= ROOT_VERSION(6,8,0)
+      MethodFastBDT( DataSetInfo& theData,
+                 const TString& theWeightFile);
+#else
       MethodFastBDT( DataSetInfo& theData,
                  const TString& theWeightFile,
                  TDirectory* theTargetDir = NULL );
+#endif
 
       /**
        * Destroys method by deleteting the forest and the transformer
diff --git a/src/MethodFastBDT.cxx b/src/MethodFastBDT.cxx
index 2754eb4..b8ef1a8 100644
--- a/src/MethodFastBDT.cxx
+++ b/src/MethodFastBDT.cxx
@@ -21,12 +21,20 @@ ClassImp(TMVA::MethodFastBDT)
 
 using namespace FastBDT;
 
+#if ROOT_VERSION_CODE >= ROOT_VERSION(6,8,0)
+TMVA::MethodFastBDT::MethodFastBDT( const TString& jobName,
+                            const TString& methodTitle,
+                            DataSetInfo& theData,
+                            const TString& theOption ) :
+   TMVA::MethodBase( jobName, Types::kPlugins, methodTitle, theData, theOption)
+#else
 TMVA::MethodFastBDT::MethodFastBDT( const TString& jobName,
                             const TString& methodTitle,
                             DataSetInfo& theData,
                             const TString& theOption,
                             TDirectory* theTargetDir ) :
    TMVA::MethodBase( jobName, Types::kPlugins, methodTitle, theData, theOption, theTargetDir )
+#endif
    , fNTrees(0)
    , fShrinkage(0)
    , fRandRatio(0)
@@ -40,10 +48,16 @@ TMVA::MethodFastBDT::MethodFastBDT( const TString& jobName,
 {
 }
 
+#if ROOT_VERSION_CODE >= ROOT_VERSION(6,8,0)
+TMVA::MethodFastBDT::MethodFastBDT( DataSetInfo& theData,
+                            const TString& theWeightFile)
+   : TMVA::MethodBase( Types::kPlugins, theData, theWeightFile)
+#else
 TMVA::MethodFastBDT::MethodFastBDT( DataSetInfo& theData,
                             const TString& theWeightFile,
                             TDirectory* theTargetDir )
    : TMVA::MethodBase( Types::kPlugins, theData, theWeightFile, theTargetDir )
+#endif
    , fNTrees(0)
    , fShrinkage(0)
    , fRandRatio(0)
